### PR TITLE
Add CbpStrategy accepting OBP params

### PR DESCRIPTION
**Pull request by ecoologic**
if you use the iterator with a CBP resource, you might not need to update the sort and pagination parameters.

This starts the upgrade guide, but please don't consider it complete or final. More work will come in that area.

I suggest reviewers to use **Split view**.

### CODE REVIEW

File: src/Zendesk/API/Traits/Utility/Pagination/CbpStrategy.php [hunk 1]
```diff
@@ -10,21 +10,64 @@ class CbpStrategy extends AbstractStrategy
...
+    private function unsetObpParams($params)
+    {
+        unset(
+            $params['page'],
+            $params['per_page'],
+            $params['sort_by'],
+            $params['sort_order']
+        );
+        return $params;
+    }
```
Comment (ecoologic) [inline src/Zendesk/API/Traits/Utility/Pagination/CbpStrategy.php:72]: so now the CbpStrategy knows a lot about OBP. I thought of extracting it, and I tried, 
but it leads to a lot of indirection. In the end I prefer to accept some duplication and imperfect encapsulation in favour of simpler code.

File: UPGRADE_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Upgrade guide
```
Comment (ecoologic) [inline UPGRADE_GUIDE.md:1]: There's a little duplication in the info below, but I think it's worth giving a TLDR summary.

File: README.md [hunk 3]
```diff
@@ -132,13 +132,31 @@ There are two ways to do pagination in the Zendesk API, **CBP (Cursor Based Pagi
 The use of the correct pagination is encapsulated using the iterator pattern, which allows you to retrieve all resources in all pages, without having to deal with pagination at all:
 
 ```php
-$ticketsIterator = $client->tickets()->iterator();
+$iterator = $client->tickets()->iterator();
 
-foreach ($ticketsIterator as $ticket) {
-    process($ticket); // Your implementation
+foreach ($iterator as $ticket) {
+    echo($ticket->id . " ");
```
Comment (ecoologic) [inline README.md:138]: I now prefer code that is fully testable, without having to define `process`

File: UPGRADE_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+* CBP: https://{subdomain}.zendesk.com/api/v2/tickets?sort=assignee.name&page[size]=2 `HTTP 400`
+
+```json
+{
+    "error": "InvalidPaginationParameter",
+    "description": "sort is not valid"
+}
+```
+
+If this is your situation, **you need to change sorting order** to a supported one.
```
Comment (Zendesk-NathanBellette) [inline UPGRADE_GUIDE.md:54]: Possibly change this to:

> If this is your situation, **you will need to change the sorting order** to a supported one.

File: UPGRADE_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+## The new iterator
+
+Your best solution to implement CBP is to use the newly provided iterator on your resources, for example:
```
Comment (Zendesk-NathanBellette) [inline UPGRADE_GUIDE.md:58]: Nitpickiest nitpick. I

> The most elegant way to implement CBP....

**Review by Zendesk-NathanBellette — APPROVED** [review]
NIce work! Approved.

**Pull request merged**